File: src/scrt/com/packet/JunctionID.java
package scrt.com.packet;

import java.io.IOException;
import java.io.InputStream;
import java.util.List;

public class JunctionID extends ID
{
	public int Number;
	public String Name;
	public JunctionID() {}
	public JunctionID(InputStream i) throws IOException
	{
		super(i);
		Number = i.read();
		Name = "A" + String.valueOf(Number);
	}
	@Override
	public List<Integer> getId()
	{
		List<Integer> l = super.getId();
		l.add(Number);
		return l;
	}
}

File: common/packet/JunctionID.cpp
#include "JunctionID.h"
JunctionID::JunctionID() : ID(ElementType::Junction) {}
JunctionID::JunctionID(char *&data) : ID(data)
{
    Number = *data++;
    Name = "A" + to_string(Number);
}
vector<char> JunctionID::getId()
{
    vector<char> l = ID::getId();
    l.push_back(Number);
    return l;
}
ID *JunctionID::byState(char *&data)
{ 
    return new JunctionID(data);
}